consensus/consensus-types/src/block.rs

```diff
@@ -361,6 +367,27 @@ impl Block {
             .map(|address| voters.contains_key(address))
             .collect()
     }
+
+    fn failed_authors_to_indices(
+        validators: &[AccountAddress],
+        failed_authors: &[(Round, Author)],
+    ) -> Vec<u32> {
+        failed_authors
+            .iter()
+            .map(|(_round, failed_author)| {
+                validators
+                    .iter()
+                    .position(|&v| v == *failed_author)
```

Comment:
Since you're newly adding this, can you use the validator index from validator config instead? https://github.com/aptos-labs/aptos-core/blob/main/types/src/validator_config.rs#L42

This can ensure that in future when we deprecate validator set sorting (expensive to do on chain), we can still ensure that this works correctly

Reply:
Turns out this is not easy to do as block execution doesn't readily have access to the validator set/validator info needed to fetch validator indices. We can do this later in a different PR when we clean up the sorting logic.